framework/src/meshgenerators/LowerDBlockFromSidesetGenerator.C

```diff
@@ -143,7 +137,10 @@ LowerDBlockFromSidesetGenerator::generate()
 
     // Add id for distributed
     if (distributed)
-      side_elem->set_id(max_elem_id + processor_id() * max_elems_to_add + i);
+    {
+      side_elem->set_id(max_elem_id + i);
+      side_elem->set_unique_id() = max_elem_id + i;
```

Comment:
How can this be unique in parallel?

Reply:
Oh - because we aren't deleting remotes yet?

Reply:
Right, and even if we do delete *some* remotes, I would have a geometric ghosting functor that ensures that all the boundary elements were available on all processes and that we would be generating these lower dimensional elements on all processes, e.g. they would be owned by one process and ghosts on all other processes. Hence since we are adding on all processes we actually want their IDs to be the same.